src/main/java/io/github/amerebagatelle/fabricskyboxes/skyboxes/AbstractSkybox.java

```diff
@@ -293,42 +292,36 @@ public void renderDecorations(WorldRendererAccess worldRendererAccess, MatrixSta
             // moon
             s = 20.0F;
             if (decorations.isMoonEnabled()) {
-                worldRendererAccess.getTextureManager().bindTexture(this.decorations.getMoonTexture());
-                int t = world.getMoonPhase();
-                int u = t % 4;
-                int v = t / 4 % 2;
-                float w = (float) (u) / 4.0F;
-                float o = (float) (v) / 2.0F;
-                float p = (float) (u + 1) / 4.0F;
-                float q = (float) (v + 1) / 2.0F;
-                bufferBuilder.begin(7, VertexFormats.POSITION_TEXTURE);
-                bufferBuilder.vertex(matrix4f2, -s, -100.0F, s).texture(p, q).next();
-                bufferBuilder.vertex(matrix4f2, s, -100.0F, s).texture(w, q).next();
-                bufferBuilder.vertex(matrix4f2, s, -100.0F, -s).texture(w, o).next();
-                bufferBuilder.vertex(matrix4f2, -s, -100.0F, -s).texture(p, o).next();
+                RenderSystem.setShaderTexture(0, this.decorations.getMoonTexture());
+                int u = world.getMoonPhase();
+                int v = u % 4;
+                int w = u / 4 % 2;
+                float x = (float)(v + 0) / 4.0F;
+                float p = (float)(w + 0) / 2.0F;
+                float q = (float)(v + 1) / 4.0F;
+                float z = (float)(w + 1) / 2.0F;
+                bufferBuilder.begin(VertexFormat.DrawMode.QUADS, VertexFormats.POSITION_TEXTURE);
+                bufferBuilder.vertex(matrix4f2, -s, -100.0F, s).texture(q, r).next();
+                bufferBuilder.vertex(matrix4f2, s, -100.0F, s).texture(x, r).next();
+                bufferBuilder.vertex(matrix4f2, s, -100.0F, -s).texture(x, p).next();
+                bufferBuilder.vertex(matrix4f2, -s, -100.0F, -s).texture(q, p).next();
                 bufferBuilder.end();
                 BufferRenderer.draw(bufferBuilder);
             }
             // stars
             if (decorations.isStarsEnabled()) {
                 RenderSystem.disableTexture();
-                float aa = world.method_23787(tickDelta) * r;
-                if (aa > 0.0F) {
-                    RenderSystem.color4f(aa, aa, aa, aa);
-                    worldRendererAccess.getStarsBuffer().bind();
-                    worldRendererAccess.getSkyVertexFormat().startDrawing(0L);
-                    worldRendererAccess.getStarsBuffer().draw(matrices.peek().getModel(), 7);
-                    VertexBuffer.unbind();
-                    worldRendererAccess.getSkyVertexFormat().endDrawing();
+                float ab = world.method_23787(tickDelta) * s;
+                if (ab > 0.0F) {
+                    RenderSystem.setShaderColor(ab, ab, ab, ab);
+                    worldRendererAccess.getStarsBuffer().setShader(matrices.peek().getModel(), matrix4f, GameRenderer.getPositionShader());
```

Comment:
I believe Vanilla now provides a `WorldRenderer#renderStars` method, is that usable here?